src/Microsoft.OpenApi/Reader/OpenApiModelFactory.cs

```diff
@@ -239,14 +234,15 @@ public static T Parse<T>(string input,
             return Load<T>(stream, version, format, openApiDocument, out diagnostic, settings);
         }
 
-        private static readonly OpenApiReaderSettings DefaultReaderSettings = new();
+        private static readonly Lazy<OpenApiReaderSettings> DefaultReaderSettings = new(() => new OpenApiReaderSettings());
 
         private static async Task<ReadResult> InternalLoadAsync(Stream input, string format, OpenApiReaderSettings settings, CancellationToken cancellationToken = default)
         {
-            var reader = OpenApiReaderRegistry.GetReader(format);
+            settings ??= DefaultReaderSettings.Value;
+            var reader = settings.Readers[format];
```

Comment:
By removing the registry, one thing I believe we may miss out on is that if a reader(the key) is absent, the error that the library user will get may not be too useful/helpful(they will get a [KeyNotFoundException](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.keynotfoundexception?view=net-9.0)).
Would it make sense to add a getReader method (can be private/internal) so as to throw a more useful error for such a scenario where the settings isn't setup right?

Reply:
Added a method to throw more explicit exceptions, which also lead me to add a lot of additional unit tests. Thanks for pointing this out.